spec/models/rental_spec.rb

```diff
@@ -223,4 +223,31 @@
     rent = create :mock_rental, item_type: create(:item_type, name: 'Test 220', base_fee: 200, fee_per_day: 20)
     expect(FinancialTransaction.where(rental: rent).map(&:amount)).to eq([220])
   end
+
+  describe '#delete_reservation' do
+    context 'error thrown' do
+      it 'logs the error and returns false' do
+        r = build(:rental)
+        r.create_reservation
```

Comment:
They do appear to be necessary; I tried to do the above code and it seems that building a rental does not trigger `create_reservation` . I've noticed while working with the unfriendly errors that the callbacks don't seem to be working the way that we've been expecting them to and am trying to fix that problem in an upcoming PR.